tests/e2e/improved_csi_idempotency.go

```diff
@@ -0,0 +1,537 @@
+		5. Verify resize was successful
+		6. Delete pvcs and SC
+		7. Verify no orphan volumes are left
+	*/
+	ginkgo.It("extend volume when csi restarts - idempotency", func() {
+		serviceName = "CSI"
+		extendVolumeWithServiceDown(serviceName, namespace, client, storagePolicyName, scParameters,
+			volumeOpsScale, true, isServiceStopped)
+	})
+
+	/*
+		Extend volume when CNS goes down
+		1. Create a SC using a thick provisioned policy
+		2. Create a PVCs using SC and wait for PVC to be bound
+		3. Extend pvcs and wait for vc task to start
+		4. Bring down vsan-health service for 6mins
+		5. Restart vsan-health service
+		6. Verify resize was successful
+		7. Delete pvcs and SC
+		8. Verify no orphan volumes are left
+	*/
+	ginkgo.It("extend volume when CNS goes down - idempotency", func() {
+		serviceName = vsanhealthServiceName
+		extendVolumeWithServiceDown(serviceName, namespace, client, storagePolicyName, scParameters,
+			volumeOpsScale, true, isServiceStopped)
+	})
+})
+
+// createVolumeWithServiceDown creates the volumes and immediatly stops the services and wait for
+// the service to be up again and validates the volumes are bound
+func createVolumeWithServiceDown(serviceName string, namespace string, client clientset.Interface,
+	storagePolicyName string, scParameters map[string]string, volumeOpsScale int, isServiceStopped bool) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ginkgo.By(fmt.Sprintf("Invoking Test for create volume when %v goes down", serviceName))
+	var storageclass *storagev1.StorageClass
+	var persistentvolumes []*v1.PersistentVolume
+	var pvclaims []*v1.PersistentVolumeClaim
+	var err error
+	var fullSyncWaitTime int
+	ignoreLabels := make(map[string]string)
+	pvclaims = make([]*v1.PersistentVolumeClaim, volumeOpsScale)
+
+	// Decide which test setup is available to run
+	if vanillaCluster {
+		ginkgo.By("CNS_TEST: Running for vanilla k8s setup")
+		// TODO: Create Thick Storage Policy from Pre-setup to support 6.7 Setups
+		scParameters[scParamStoragePolicyName] = "Management Storage Policy - Regular"
+		// Check if it is file volumes setups
+		kind := os.Getenv("ACCESS_MODE")
+		if strings.TrimSpace(string(kind)) == "RWX" {
+			scParameters[scParamFsType] = nfs4FSType
+		}
+		storageclass, err = createStorageClass(client, scParameters, nil, "", "", false, "idempotency")
+	} else if supervisorCluster {
+		ginkgo.By("CNS_TEST: Running for WCP setup")
+		thickProvPolicy := os.Getenv(envStoragePolicyNameWithThickProvision)
+		if thickProvPolicy == "" {
+			ginkgo.Skip(envStoragePolicyNameWithThickProvision + " env variable not set")
+		}
+		profileID := e2eVSphere.GetSpbmPolicyID(thickProvPolicy)
+		scParameters[scParamStoragePolicyID] = profileID
+		// create resource quota
+		createResourceQuota(client, namespace, rqLimit, thickProvPolicy)
+		storageclass, err = createStorageClass(client, scParameters, nil, "", "", false, thickProvPolicy)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
```

Comment:
This check is made in line 263. You can remove this

Reply:
removed